### ISSUE TITLE
V2 Production Emails sent to potential CLA Manager and Company Admin from Contributor are wrong

### ISSUE THREAD

**Issue by WillsonHG** [communitybridge/easycla#2427]
## Summary
The emails sent to a potential cla manager or the company admin have not been proofed and are inconsistent.

## Background

1. A contributor is gated by easycla
2. The contributor chooses CCLA
3. The contributor's company has not signed the CCLA for the project
4. The contributor chooses to send an email to a CLA manager OR the Company Admin.

Emails sent to company admin:
![image](https://user-images.githubusercontent.com/46732999/104198678-809f9600-53f4-11eb-8b03-a86049204527.png)
![image](https://user-images.githubusercontent.com/46732999/104198752-96ad5680-53f4-11eb-98de-2aaa0872fb68.png)

Emails sent to CLA Manager Designee
![image](https://user-images.githubusercontent.com/46732999/104198864-c0ff1400-53f4-11eb-9019-772aacb55320.png)
![image](https://user-images.githubusercontent.com/46732999/104198930-d4aa7a80-53f4-11eb-8b7a-e5a579c21ae9.png)

The logic for when these emails are sent needs to be reviewed, as well as the content of the emails themselves that have incorrect grammar and broken links.

**Comment by vinod-kadam** [comment]
As updated by @wanyaland , the following content (which is used to send emails) was shared by @pranab-bajpai  on slack however it seems the email document was not updated with the new content.
-------------------------
Subject: "EasyCLA: Invitation to Sign the {company_name} Corporate CLA"
This is a notification email from EasyCLA regarding the CLA setup and signing process for {company_name}.
{user} <{email}> has identified you as a potential candidate to setup the Corporate CLA for {company_name} in support of the following projects:
{project A name}
{project B name}
…
Before any code contribution can be accepted, your organization must sign a CLA. Either you or someone whom you designate from your company can login to this portal <link> and sign the CLA for this project <project_name>. Please notify the user once CLA setup is complete so that they may complete the code contribution process.
{cla help notice….}
{cla signoff} 
-------------------------

**Comment by vinod-kadam** [comment]
As updated by @pranab-bajpai , the above content (given by Harold) should go in the document.

**Comment by vinod-kadam** [comment]
David has pushed API changes to PROD so now @WillsonHG you can verify emails.

**Comment by WillsonHG** [comment]
As a contributor - I entered an email for a CLA Manager that was not in the system - hwillson+clamanager2@contractor.linuxfoundation.org 

I received 2 emails to that address.
![image](https://user-images.githubusercontent.com/46732999/104664888-5a048800-569e-11eb-8a43-f9c6df606352.png)

and 
![image](https://user-images.githubusercontent.com/46732999/104664929-6e488500-569e-11eb-8fad-d5e73a564137.png)

**Comment by WillsonHG** [comment]
No Email was sent to the Company Admin

**Comment by WillsonHG** [comment]
still the same email sent to hwillson+designee
![image](https://user-images.githubusercontent.com/46732999/104667209-8ec70e00-56a3-11eb-8ce1-10de0ee4eb7f.png)

**Comment by thakurveerendras** [comment]
Issue is tested on staging environment & found only below email after performing steps mentioned in descriptions.

![image.png](https://images.zenhubusercontent.com/5fc8abbc87345c4b3bd7aace/f7936578-2f90-47d2-94cc-fc9ea19ec5e0)

**Refer video clip for more details :** https://images.zenhubusercontent.com/194341141/543893d5-223c-40e2-a030-073b8a84b0e0/easycla2427.mp4

**Comment by vinod-kadam** [comment]
Corporate Console link need to be corrected to member.lfx.linuxfoundation.org.

**Comment by manishv09** [comment]
It's reproduced when no CLA is signed for an Organization and user selects Identify CLA Manager as No and proceed to give CLA Manager information and submit the request.
2 different emails are triggered with subject line as follows:
1. EasyCLA: Invitation to create LF Login and complete process of becoming CLA Manager & Email content as follows: 
![image.png](https://images.zenhubusercontent.com/5face8c0a2ad18653594316c/b2f6d90e-2092-4b91-a0e0-9158aae53125)

2. EasyCLA: Invitation to create LF Login and complete process of becoming CLA Manager with cla-manager-designee role & Email content as follows:
![image.png](https://images.zenhubusercontent.com/5face8c0a2ad18653594316c/9b094d53-c4b1-458c-b111-f73a3dba46f0)

**Comment by vinod-kadam** [comment]
As discussed in 19-Jan standup, there are two use cases mentioned in this ticket:

1. Contributor designates someone (who has LFID) as a cla-manager to sign CCLA 
-->  designated cla-manager receives two emails:  bit different subject line, one email has more user contributor information then the other
2. Contributor designates someone (who doesn't have LFID) as a cla-manager to sign CCLA --> designated cla-manager receives only one email which is correct however the email need to be updated to add contributor information.

**Comment by pranab-bajpai** [comment]
For the first case Use Case 1: Contributor designates someone (who doesn't have LFID) as a cla-manager to sign CCLA --> designated cla-manager receives only one email which is correct however the email need to be updated to add contributor information. Below email is fine
![image.png](https://images.zenhubusercontent.com/5ee7d7b3dcd351d6fe0c206c/2d4e7bf1-fabb-4bf6-85b1-488fe87b9925)
For the second case Use Case 2 (low priority): Contributor designates someone (who has LFID) as a cla-manager to sign CCLA
--> designated cla-manager receives two emails: bit different subject line, one email has more user contributor information then the other. Need to remove the email which has less contributor information. Only one email like the one below should be sent
![image.png](https://images.zenhubusercontent.com/5ee7d7b3dcd351d6fe0c206c/8b630c3a-2f08-4b6c-a8d4-e34b4999677a)

**Comment by wanyaland** [comment]
Updated and changes have been pushed to DEV and STAGING
- on email content for users that have no LFID 
- removed the second email that was being sent
For the use case where users have LFID content was as expected but can adjust based on feedback
@dealako , @vinod-kadam , @WillsonHG

**Comment by WillsonHG** [comment]
I tested this evening on company TEST and sent to an email not in the system and received the following email:
![image](https://user-images.githubusercontent.com/46732999/106223160-e8a0fb00-61ae-11eb-8373-ccf76eb1ec05.png)

This email needs a proof for grammar and clarity.
It also has the project and organization swapped in the first line.


@pranab-bajpai  can you please link the master email list here?

**Comment by WillsonHG** [comment]
The email sent to a cla manager with a user in the system.

![image](https://user-images.githubusercontent.com/46732999/106224275-053e3280-61b1-11eb-8e7c-6110999696fd.png)

This one also needs a proof for grammar.
The link to the corporate console is still wrong.

**Comment by WillsonHG** [comment]
We also need to decide "who" these emails come from. One says "admin" which is too vague and the other is "noreply".

![image](https://user-images.githubusercontent.com/46732999/106224383-46364700-61b1-11eb-857a-4e2a9abb525b.png)

**Comment by vinod-kadam** [comment]
As updated by @manishv09, use case are working fine on Dev - was tested on Dev as blocker fix LFX-3058 was available only on Dev.

Use Case 1: Contributor designates someone (who doesn't have LFID) as a cla-manager to sign CCLA --> designated cla-manager receives only two emails; the email need to be updated to add contributor information : Deployed to Dev, Staging & PROD: QA Verified on Staging

Use Case 2 (low priority): Contributor designates someone (who has LFID) as a cla-manager to sign CCLA
--> designated cla-manager receives two emails: bit different subject line, one email has more user contributor information then the other. Need to remove the email which has less contributor information. : Deployed to Dev, Staging & PROD: was verified on Dev as blocker fix LFX-3058 was available only on Dev - working fine.

So, moved the ticket to QA Done.

**Comment by wanyaland** [comment]
Have refactored the link for corporate console -v2 and have pushed changes to dev. 
cc @vinod-kadam

**Comment by vinod-kadam** [comment]
@Heather, this is now available on PROD. So, please verify it on PROD.

**Comment by manishv09** [comment]
Verified on Staging environment and found the email is received as follows:
![image.png](https://images.zenhubusercontent.com/5face8c0a2ad18653594316c/daf81806-6179-4505-bec8-a73d64361f60)

**Comment by vinod-kadam** [comment]
@WillsonHG  please verify it on PROD.

**Comment by WillsonHG** [comment]
This ticket was made to address 3 email scenarios from the CONTRIBUTOR console. All three scenarios are still failing. 

BACKGROUND:

There is a company in the system, but they have not signed the CCLA for the project. The contributor is sending an email to either the company admin or someone they think is the cla manager designee. There are 2 actions for the cla manager - 1) Sign the cla and 2) add the contributor to the approved list. None of the emails in the document describe this process accurately, nor do either of the 2 emails listed by Pranab earlier in this ticket.

1) The contributor chooses to send an email to the company admin. - FAIL, no email sent. https://github.com/communitybridge/easycla/issues/2634


2) The contributor chooses to send an email to a cla manager that they designate. That cla manager does NOT have an LF Account with the email provided by the contributor. - FAIL The following email was received in error. Do not send this email out.
![image](https://user-images.githubusercontent.com/46732999/107997645-3c825100-6fb1-11eb-9c41-d7ad0b211e3a.png)


3) The contributor chooses to send an email to a cla manager that they designate. That cla manager DOES have an LF Account with the email provided by the contributor. - FAIL - The email sent is not ready for production, and the link to the Member portal is still wrong.
![image](https://user-images.githubusercontent.com/46732999/108000442-fbda0600-6fb7-11eb-8c09-d676794f7982.png)

I have edited the above email for grammar and clarity.

Hello Heather Designee,

This is a notification email from EasyCLA regarding the project PROJECT.

The following contributor would like to contribute to PROJECT on behalf of your organization:

hwccla (willsonhg2+ccla@gmail.com)

Before the user contribution can be accepted, your organization must sign a CLA.

Kindly login to this portal https://member.lfx.linuxfoundation.org/ and sign the CLA for the project PROJECT.

After signing the CLA, you will need to add this contributor to the approved list. Please notify the contributor once they are added, so that they may complete the contribution process.

If you need help or have questions about EasyCLA, you can read the documentation or reach out to us for support.

EasyCLA Support Team

**Comment by dealako** [comment]
Pushed to DEV - I will QA

**Comment by vinod-kadam** [comment]
@makkalot, David has pushed the email updates (rewording - he worked with Heather for it) to DEV/STAGING but the accept invite link was removed. We need to reintroduce the invite link as discussed yesterday since it is needed.

**Comment by vinod-kadam** [comment]
@dealako  & @WillsonHG  please review this - this is what I provided to Denis to use.
<p>Hello {{.RecipientName}},</p>
<p>This is a notification email from EasyCLA regarding the project {{.GetProjectNameOrFoundation}}.</p>
<p>The following contributor would like to contribute to {{.GetProjectNameOrFoundation}} on behalf of your organization: {{.CompanyName}}.</p>
<p>{{.RequesterUserName}} ({{.RequesterEmail}})</p>
<p>Before the user's contribution can be accepted, your organization must sign a CLA.</p>
<p>Please click on <a href="USERACCEPTLINK">Accept Invite</a> to create your LF Login.</p>
<p>After login, you will be redirected to this portal {{.CorporateConsole}} where you can sign the CLA for the project {{.GetProjectNameOrFoundation}}.</p>
<p>After signing the CLA, you will need to add this contributor to the approved list. Please notify the contributor once they are added, so that they may complete the contribution process.</p>

**Comment by vinod-kadam** [comment]
The above change was approved by David over slack.
The change has been deployed to DEV, Staging & PROD.

**Comment by manishv09** [comment]
Verified on Staging and found the changes mentioned on above by @vinod-kadam, @makkalot and approved by @dealako is as follows:

![image.png](https://images.zenhubusercontent.com/5face8c0a2ad18653594316c/bd79d89a-2478-43ee-b376-2ec386c323d6)
